### ISSUE TITLE
Questions regarding Second Stage Loss Targets

### ISSUE THREAD

**Issue by anuj-sharma-19** [tianweiy/CenterPoint#363]
Hi,

Thanks again for your great work!

I am working on the Second Stage of CenterPoint. In the code, I see the score target to be `IOU3D score > fg_thresh` https://github.com/tianweiy/CenterPoint/blob/08c973a5054ac8dadae2e9dc54e9109123e8076b/det3d/models/roi_heads/target_assigner/proposal_target_layer.py#L60. However, in the paper, the class target score is 

```
I = min(1, max(0, 2 × IoUt − 0.5))
```

Could you please clarify on which one I should use here?

Thanks!

**Comment by tianweiy** [comment]
hi, one line 61, the cls label is computed as 
```
batch_cls_labels[interval_mask] = \
                (batch_roi_ious[interval_mask] - iou_bg_thresh) / (iou_fg_thresh - iou_bg_thresh)```

where bg_thresh is 0.25 and fg_thresh is 0.75. 

So mathematically, this matches the description in the original paper. 

Best

**Comment by anuj-sharma-19** [comment]
Hi @tianweiy thank you for the clarification. Makes sense :+1: 

Cheers,
Anuj

**Comment by anuj-sharma-19** [comment]
Hi @tianweiy I was wondering if you had any results on nuScenes with Second Stage that you could please share? It would be helpful to understand what performance to expect from it. Also, if you have any training config for this on nuScenes, it would be really great.

Thank you!

**Comment by tianweiy** [comment]
see https://arxiv.org/abs/2111.06881 

you can get about 0.5 nds improvement using simplest bev feature aggregation. AFAIK, fancier two-stage approach that involves raw point cloud doesn't really work on nuScenes (similar improvement in comparison to bev feature)

https://github.com/tianweiy/CenterPoint/blob/master/configs/mvp/nusc_two_stage_base_with_virtual.py
this config is for two stage with virtual points. You can modify it to work with non-virtual point cases. You will need to download a pretrained single stage model for initialization.

**Comment by anuj-sharma-19** [comment]
Thanks for your quick response!

Is the Two-Stage approach not the same as BEV Feature aggregation?

**Comment by tianweiy** [comment]
it is the same
